server/routerlicious/packages/services/src/mongodb.ts

```diff
@@ -168,21 +169,18 @@ export class MongoCollection<T> implements core.ICollection<T>, core.IRetryable
         }
     }
 
-    public async findOrCreate(query: any, value: T): Promise<{ value: T; existing: boolean; }> {
+    public async findOrCreate(query: any, value: any): Promise<{ value: T; existing: boolean; }> {
         const req = async () => {
             const result = await this.collection.findOneAndUpdate(
-                query,
-                {
-                    $setOnInsert: value,
-                },
-                {
-                    returnOriginal: true,
-                    upsert: true,
-                });
+                {query},
+                {$setOnInsert: value},
+                {upsert: true,
+                returnDocument: "before"}
+            );
 
             return result.value
-                ? { value: result.value, existing: true }
-                : { value, existing: false };
+                ? { value: result.value, existing: true } as { value: T; existing: boolean; }
+                : { value, existing: false } as { value: T; existing: boolean; };
```

Comment:
`value` was `T`, and by changing to `any` you have to cast it back to `T`. Any reason why we do that?